src/Interpreters/ActionsDAG.h

```diff
@@ -164,7 +164,7 @@ class ActionsDAG
         const FunctionBasePtr & function_base,
         NodeRawConstPtrs children,
         std::string result_name);
-    const Node & addCast(const Node & node_to_cast, const DataTypePtr & cast_type, std::string result_name);
+    const Node & addCast(const Node & node_to_cast, const DataTypePtr & cast_type, std::string result_name, ContextPtr context = nullptr);
```

Comment:
And here again, don't use nullptr as default value, we should always propagate context to this and `makeConvertingActions` methods. Places where we need `nullptr` as a context should not exist (or be rare exceptions with a reason for this).